static/js/common/fxa-login.js

```diff
@@ -10,42 +10,62 @@
         "clientId": "cd5a21fafacc4744",
         "contentHost": "https://stable.dev.lcip.org",
         "loginUri": "http://olympia.dev/api/v3/accounts/login/",
+        "registerUri": "http://olympia.dev/api/v3/accounts/register/",
         "oauthHost": "https://oauth-stable.dev.lcip.org/v1",
         "redirectUri": "http://olympia.dev/fxa-authorize",
         "scope": "profile",
     };
     var fxaClient = getClient();
 
-    function fxaLogin() {
-        console.log('[FxA] Starting sign in');
-        return fxaClient.auth.signIn({
+    function fxaLogin(opts) {
+        function postConfig(response) {
+            return {
+                action: response.action,
+                code: response.code,
+                state: response.state,
+                client_id: config.clientId,
+            };
+        }
+
+        opts = opts || {};
+        var authConfig = {
             ui: 'lightbox',
             state: 'foo',
             redirectUri: config.redirectUri,
             scope: config.scope,
-        });
+        };
+        if (opts.signUp) {
+            console.log('[FxA] Starting sign up');
+            return fxaClient.auth.signUp(authConfig).then(function(response) {
+                console.log('[FxA] Register success', response);
+                return $.post(config.registerUri, postConfig(response));
+            });
+        } else {
+            console.log('[FxA] Starting sign in');
+            return fxaClient.auth.signIn(authConfig).then(function(response) {
+                console.log('[FxA] Login success', response);
+                return $.post(config.loginUri, postConfig(response));
+            });
+        }
     }
 
     $('body').on('click', '.fxa-login', function(e) {
         e.preventDefault();
 
         fxaLogin().then(function(response) {
-            console.log('[FxA] Login success', response);
-            var headers = new Headers();
-            $.post(config.loginUri, {
-                action: response.action,
-                code: response.code,
-                state: response.state,
-                client_id: config.clientId,
-            }).then(function(response) {
-                console.log('[FxA] Server login response', response);
-                window.location.reload();
-            }, function(error) {
-                console.log('[FxA] Server login error', error);
-                alert('There was an error logging you in');
-            });
-        }, function() {
-            console.log('[FxA] Login failed', arguments);
+            console.log('[FxA] Server login response', response);
+            window.location.reload();
+        }, function(error) {
+            console.log('[FxA] Login failed', error);
+        });
+    }).on('click', '.fxa-register', function(e) {
+        e.preventDefault();
+
+        fxaLogin({signUp: true}).then(function(response) {
+            console.log('[FxA] Server register response', response);
```

Comment:
Yeah, I think it will move to a django view that logs the user in. We'll still need the API for when we switch to a React app though so I'll update these views later.